Tests/Api/Parameter/EnumParameterTest.php

```diff
@@ -72,4 +72,34 @@ public function testFoundValueInEnum(): void
 
         $this->assertTrue($parameter->isValid());
     }
+
+    /**
+     * @covers \Chaplean\Bundle\ApiClientBundle\Api\Parameter::enum
+     * @covers \Chaplean\Bundle\ApiClientBundle\Api\Parameter\EnumParameter::__construct
+     * @covers \Chaplean\Bundle\ApiClientBundle\Api\Parameter\EnumParameter::validate
+     *
+     * @return void
+     */
+    public function testValidStrictType(): void
+    {
+        $parameter = Parameter::enum([1, 2]);
+        $parameter->setValue('1');
+
+        $this->assertFalse($parameter->isValid());
+    }
+
+    /**
+     * @covers \Chaplean\Bundle\ApiClientBundle\Api\Parameter::enum
+     * @covers \Chaplean\Bundle\ApiClientBundle\Api\Parameter\EnumParameter::__construct
+     * @covers \Chaplean\Bundle\ApiClientBundle\Api\Parameter\EnumParameter::validate
+     *
+     * @return void
+     */
+    public function testArrayListEnum(): void
```

Comment:
Test name is not clear. Something more specific about what it does.
Furthermore, is there an opposite test of this one? You have a valid case, but what about not valid cases. Is that TDD?

Reply:
It would rather be integration test rather than unit tests. The TDD tests of the feature are above. This tests how the new feature behaves when use along with other existing features (here an arrayList containing the new type). But I agree about the invalid test in the context of arrayList(enum).

Reply:
But first thing to do : try the simpler case in TDD. The simpler case is: we have an empty Enum in ArrayList, it's false. Not functional, just needed. With only the test above, the code could always return true, it works...